contracts/marketplace/Marketplace.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.0;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "../DebtToken.sol";
+
+contract Marketplace {
+  using SafeMath for uint256;
+  using SafeERC20 for DebtToken;
+
+  //Structure for sell position
+  struct sellPosition{
+    uint256 amount; //tokens that can be selled
+    uint256 sellPrice; //sell price per token in wei
+    uint256 minAmountToBuy; //minimum amount of tokens for someone to buy
+    bool    exists;
+  }
+
+  //Map of selling positions, quantity -> sellPrice(wei)
+  mapping(address => mapping( string => sellPosition)) sellPositions;
+
+  function createSellPosition(DebtToken token, uint256 _sellPrice, uint256 _minAmountToBuy, uint256 _amount) public {
```

Comment:
Shouldn't trust user to provide correct, non-malicious ERC-20 contract.

Reply:
Is a DebtToken token, only the company(us) should have permissions to generate it.

Reply:
User can nevertheless provide contract generated by himself.

Reply:
We can prevent that by checking the deployer of the DebtToken?

Reply:
AFAIK, there is no secure way to retrieve that information.

Reply:
We can have a list of token DebtToken contracts that we allow to be in the marketplace, and a admin method to add and remove more DebtToken contracts to that list.

Reply:
Probably ok. May also be interesting to consider cryptographic signatures to prove contract authorship.